packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -192,42 +192,35 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
     }
 
     /**
-     * {@inheritDoc (ISharedObject:interface).summarize}
+     * {@inheritDoc (ISharedObject:interface).captureSummaryState}
      */
-    public summarize(fullTree: boolean = false, trackState: boolean = false): ISummaryTreeWithStats {
-        // Set _isSummarizing to true. This flag is used to ensure that we only use SummarySerializer (created below)
-        // to serialize handles in this object's data. The routes of these serialized handles are outbound routes
-        // to other Fluid objects.
-        assert(!this._isSummarizing, 0x076 /* "Possible re-entrancy! Summary should not already be in progress." */);
-        this._isSummarizing = true;
```

Comment:
I didn't see any need to prevent the use of `summarizer` during regular summarization and changed it to only check for if we're garbage collecting.

Reply:
Can you please glance at history of when this was added / what it was supposed to protect against? And is it still valid to assert it even for smaller scope (GC)? I think it does, but would be nice to update comments to clearly articulate why reentrancy is bad / what will break if reentranacy were to happen

Reply:
Here you go: https://github.com/microsoft/FluidFramework/pull/3975/files#diff-97e77be555c966d15b3e7d746dbbc24daf3e5950606ff395cc88638119f40620R93

Reply:
Based on the history, I see a series of changes where getting GC data during summarization was more entangled and then some of it removed without also removing the protection of _isSummarizing and usage of SummarySerializer. My interpretation of the current code is that during GC (which calls summarization code), we still want to prevent the use of any other serializer except the SummarySerializer passed in. But during real summarization, we don't care about this at all. And we don't otherwise care about re-entrancy except inasmuch as it messes with the value of _isSummarizing/GCing. @agarwal-navin, is this correct?

Reply:
Yeah, that sounds about right. We need to use the `SummarySerializer` during GC because we want to get a list of all handles serialized during that process. Since summarize is done after getting the GC data, it's fine to remove this during summarize.

Reply:
I update the comments related to _isGCing to clarify what it's for and why re-entrancy is a problem.